service/worker/addsearchattributes/workflow.go

```diff
@@ -105,6 +106,8 @@ func AddSearchAttributesWorkflow(ctx workflow.Context, params WorkflowParams) er
 	logger := workflow.GetLogger(ctx)
 	logger.Info("Workflow started.", tag.WorkflowType(WorkflowName))
 
+	ctx = workflow.WithTaskQueue(ctx, primitives.AddSearchAttributesActivityTQ)
```

Comment:
The plan is to keep the default activity worker in the next release and remove it in the one after. By that point the existing task queues should have been drained, so I don't think there will be an issue.

We will run our upgrade/downgrade test pipeline before deploying to be sure.

Reply:
Huh.. the fact that WithActivityOptions doesn't override this was not obvious to me at all. I kinda feel like the tq should just go in all of the ActivityOptions. But I do see the convenience of doing it this way. So no recommendation, just a note

Reply:
Yeah the comment on `WithActivityOptions` is a little misleading since it says all activity options will be overridden, but I did confirm the activity task queue name gets inherited. I can see the argument for making it explicit on every activity, but I thought it would get a little redundant and wanted to make it easier to update in the future.